source/docs/v3/tenancy-bootstrappers.blade.md

```diff
@@ -29,31 +29,74 @@ php artisan cache:clear --tag=tenant_123
+// E.g. Storage::disk('local')->path('') will return "/$path_to_your_application/storage/tenant42/app"
+// (Given a suffix_base of 'tenant' and a tenant with an ID of 42. Same as in the example above in the Storage path helper section)
+'root_override' => [
+    'local' => '%storage_path%/app/',
+    'public' => '%storage_path%/app/public/',
+],
+```
 
-    // other methods in the TenancyServiceProvider
+To make the tenant-aware `Storage` facade work with a custom disk, add the disk's name to `config('tenancy.filesystem.disks')` and override its root in `config('tenancy.filesystem.root_override')` as shown above.
 
-    public function boot()
-    {
-        // update the middleware used by the asset controller
-        TenantAssetsController::$tenancyMiddleware = InitializeTenancyByDomainOrSubdomain::class;
-    }
+### Assets
+
+The filesystem bootstrapper makes the `asset()` helper link to the files *of the currently initialized and identified tenant*. By default, the bootstrapper makes the helper output a URL pointing to the TenantAssetsController (`/tenancy/assets/...`), which returns a file response:
+
+```php
+// TenantAssetsController
+return response()->file(storage_path('app/public/' . $path));
+```
+
+The package expects the assets to be stored in your tenant's `app/public/` directory. For global assets (non-private assets shared among all tenants or JS/CSS assets), you may want to create a disk and use URLs from that disk instead. For example:
+
+```php
+Storage::disk('app-public')->url('tenants/logos/' . tenant()->id . '.png');
 ```
 
-This bootstrapper is the most complex one, by far. We will have a — better written — explanation in v3 docs soon, but for now, refer to the 2.x docs for information about filesystem tenancy. [https://tenancyforlaravel.com/docs/v2/filesystem-tenancy/](https://tenancyforlaravel.com/docs/v2/filesystem-tenancy/)
+To access the global assets, you can use `global_asset()` and `mix()`.
+
+The `asset()` helper's behavior can be changed by configuring the asset URL (`ASSET_URL` in your `.env`) – when the asset URL is not null, the bootstrapper will suffix the configured asset URL (the same way as `storage_path()` gets suffixed), and make the `asset()` helper output that.
 
-If you don't want to bootstrap filesystem tenancy in this way, and want to — for example — provision an S3 bucket for each tenant, you can absolutely do that. Look at the package's bootstrappers to get an idea of how to write one yourself, and feel free to implement it any way you want.
+You can disable tenancy of `asset()` in the config (`tenancy.filesystem.asset_helper_tenancy`) and explicitly use `tenant_asset()` instead. `tenant_asset()` returns your website's URL suffixed by `tenancy/assets/$nameOfYourAsset` (e.g. `https://your-site.test/tenancy/assets/asset-name.txt`). You may want to do that if you're facing issues using a package that utilizes `asset()` inside the tenant app.
```

Comment:
1. I'd change `returns your website's URL suffixed by` to `**always** returns a path to the TenantAssetController: tenant_asset('foo.txt') returns your-site.com/tenancy/assets/foo.txt` (and wrap the URL in `, I couldn't do that here due to formatting limitations)
2. Could you explain what you mean by the `You may want to do that if you're facing issues using a package that ...` point?

Reply:
> Could you explain what you mean by the You may want to do that if you're facing issues using a package that ... point?

If I'm using a package that uses `asset()` e.g. to get non-tenant-specific assets, the package won't get the asset paths correctly because of tenancy, so I'd want to disable `asset()` tenancy and use `tenant_asset()` instead in the app

Reply:
https://github.com/stancl/tenancy-docs/pull/201/commits/2aa7e44d4368152adce590c4e8403dd54fdb5eb7

Reply:
Re: 2) I see, that makes sense. When I was reading this yesterday I read only the second part after writing the review for the first sentence, so it seemed that this only says that you can use `tenant_asset()` if you're using 3rd party packages, but when you read the paragraph in full it does make sense since it's in reference to disabling the asset helper tenancy.

You can mark this as resolved 👍🏻